ui/src/contexts/IssuerState.tsx

```diff
@@ -0,0 +1,79 @@
+import { Space, Typography, message } from "antd";
+import {
+  PropsWithChildren,
+  createContext,
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from "react";
+
+import { APIError } from "src/adapters/api";
+import { getStatus } from "src/adapters/api/issuer-state";
+import { useEnvContext } from "src/contexts/Env";
+import { AsyncTask } from "src/utils/async";
+import { isAbortedError, makeRequestAbortable } from "src/utils/browser";
+
+interface IssuerState {
+  notifyChange: () => Promise<void>;
+  refreshStatus: () => Promise<void>;
+  status: AsyncTask<boolean, APIError>;
+}
+
+const IssuerStateContext = createContext<IssuerState>({
+  notifyChange: () => Promise.reject("The state context is not yet ready"),
+  refreshStatus: () => Promise.reject("The state context is not yet ready"),
+  status: { status: "pending" },
+});
+
+export function IssuerStateProvider(props: PropsWithChildren) {
+  const env = useEnvContext();
+  const [status, setStatus] = useState<AsyncTask<boolean, APIError>>({ status: "pending" });
+
+  const refreshStatus = useCallback(
+    async (signal?: AbortSignal) => {
+      const response = await getStatus({ env, signal });
+
+      if (response.isSuccessful) {
+        setStatus({ data: response.data.pendingActions, status: "successful" });
+      } else {
+        if (!isAbortedError(response.error)) {
+          void message.error(response.error.message);
+        }
+      }
+    },
+    [env]
+  );
+
+  const notifyChange = useCallback(() => {
```

Comment:
Currently, this does not inform the user about any type of pending to-be-published change. It's very specific to revocation changes. If this is expected to remain specific, I'd rename it as `notifyPendingRevocationChange`.

I suspect though that this msg will be also required to be shown when we issue new direct credentials. In such a case, I'd keep the name as it's now but require the msg `Revocation requires issuer state to be published` to be passed as a param.

Reply:
I'd remove this and reuse `refreshStatus` then